mateweather/src/mateweather-applet.c

```diff
@@ -143,25 +144,23 @@ static void place_widgets (MateWeatherApplet *gw_applet)
     }
 
     /* Create the weather icon */
-    icon_name = weather_info_get_icon_name (gw_applet->mateweather_info);
-    gw_applet->image = gtk_image_new_from_icon_name (icon_name, GTK_ICON_SIZE_BUTTON);
-
-    if (icon_name != NULL) {
+    if ((gw_applet->mateweather_info) && ((icon_name = weather_info_get_icon_name (gw_applet->mateweather_info)) != NULL)) {
+        gw_applet->image = gtk_image_new_from_icon_name (icon_name, GTK_ICON_SIZE_BUTTON);
         gtk_widget_show (gw_applet->image);
         gtk_widget_get_preferred_size (gw_applet->image, &req, NULL);
         if (horizontal)
             total_size += req.height;
         else
             total_size += req.width;
+    } else {
+        gw_applet->image = gtk_image_new ();
     }
 
-    /* Create the temperature label */
-    gw_applet->label = gtk_label_new("0\302\260F");
-
     /* Update temperature text */
-    temp = weather_info_get_temp_summary(gw_applet->mateweather_info);
-    if (temp)
-        gtk_label_set_text(GTK_LABEL(gw_applet->label), temp);
+    if ((gw_applet->mateweather_info != NULL) && ((temp = weather_info_get_temp_summary (gw_applet->mateweather_info)) != NULL))
+        gw_applet->label = gtk_label_new (temp);
+    else
+        gw_applet->label = gtk_label_new(_("?ºF"));
```

Comment:
Can we drop the `ºF` from here? If I have my temp in Celsius, when re-loading the applet in the panel there's a brief period in which it shows `?ºF`

Also, note the screenshot: it shows as a smaller text than the normal temp should be, and the degrees sign has a line under it. Perhaps we should revert it back to using `gtk_label_set_text` and continue using the unicode character codes?
![Screenshot at 2020-12-14 09-53-58](https://user-images.githubusercontent.com/259780/102096403-a84e0f00-3df2-11eb-8550-9cc6afb93ea7.png)
![Screenshot at 2020-12-14 09-55-22](https://user-images.githubusercontent.com/259780/102096404-a8e6a580-3df2-11eb-8eaf-31d9d560c525.png)

Reply:
@vkareh done

Reply:
At first glance, I think the text size is smaller on your screenshot because the applet uses two rows for displaying the weather info. Perhaps we can display some icon when it cannot be got from the current weather info.

Reply:
done
```diff
diff --git a/mateweather/src/mateweather-applet.c b/mateweather/src/mateweather-applet.c
index 2048bfc8..ca07d686 100644
--- a/mateweather/src/mateweather-applet.c
+++ b/mateweather/src/mateweather-applet.c
@@ -146,15 +146,15 @@ place_widgets (MateWeatherApplet *gw_applet)
     /* Create the weather icon */
     if ((gw_applet->mateweather_info) && ((icon_name = weather_info_get_icon_name (gw_applet->mateweather_info)) != NULL)) {
         gw_applet->image = gtk_image_new_from_icon_name (icon_name, GTK_ICON_SIZE_BUTTON);
-        gtk_widget_show (gw_applet->image);
-        gtk_widget_get_preferred_size (gw_applet->image, &req, NULL);
-        if (horizontal)
-            total_size += req.height;
-        else
-            total_size += req.width;
     } else {
-        gw_applet->image = gtk_image_new ();
+        gw_applet->image = gtk_image_new_from_icon_name ("weather-storm", GTK_ICON_SIZE_BUTTON);;
     }
+    gtk_widget_show (gw_applet->image);
+    gtk_widget_get_preferred_size (gw_applet->image, &req, NULL);
+    if (horizontal)
+        total_size += req.height;
+    else
+        total_size += req.width;
 
     /* Update temperature text */
     if ((gw_applet->mateweather_info != NULL) && ((temp = weather_info_get_temp_summary (gw_applet->mateweather_info)) != NULL))
```